### DISCUSSION TITLE
The design bugs when I extend MediaResource

### DISCUSSION THREAD

**Discussion by MallauryGerard** [awcodes/filament-curator#535] [Q&A]
Hello,

I am facing the same issue that #507 

This issue appears when I extend MediaResource to modify getEloquentQuery this way 
```
class OverridedMediaResource extends MediaResource
{
    public static function getEloquentQuery(): Builder {
        return parent::getEloquentQuery()->where('directory', self::getPath());
    }

    private static function getPath() {
        $generator = config('curator.path_generator');
        return App::make($generator)->getPath(config('curator.directory'));
    }
}
```
```
// Config file
'resource' => App\Filament\Resources\OverridedMediaResource::class,
```

I created a custom theme and evething works like a charm if I use the official MediaResource...
Do you have any idea?

Thanks

**Comment by awcodes** [comment]
I wonder if there is something weird going on with filament's auto-discovery since you are putting your custom resource in it's auto-discover paths. Maybe it's getting registered twice.

**Comment by afsharmn** [comment]
same here ! :/

when i using 
`->resource(\App\Filament\Resources\CustomMediaResource::class)` in plugins registeration or 
`'resource' => \App\Filament\Resources\CustomMediaResource::class,` in config file ,
 the grid layout is not woking correctly.
 
 
![Screenshot 2024-11-18 134903](https://github.com/user-attachments/assets/7f7cc7b8-13b3-473f-8c52-626937a65b14)

and this is my CustomMediaResource :


```
<?php

namespace App\Filament\Resources;

use Awcodes\Curator\Resources\MediaResource;

class CustomMediaResource extends MediaResource
{

}
```
 how can i fix this ?
